Mongo_Adapter/AdapterActions/Pull.cs

```diff
@@ -45,7 +46,21 @@ public override IEnumerable<object> Pull(IRequest query, PullType pullType = Pul
 
             // Get the results
             List<BsonDocument> pipeline = new List<BsonDocument>();
-            if (query is BatchRequest)
+
+            if (query is CollectionNames)
+                if (m_Database == null)
+                    return null;
+                else
+                {
+                    List<string> collectionNames = new List<string>();
+                    foreach (BsonDocument collectionName in m_Database.ListCollectionsAsync().Result.ToListAsync<BsonDocument>().Result)
+                    {
+                        string name = collectionName["name"].AsString;
+                        collectionNames.Add(name);
+                    }
+                    return collectionNames;
+                }
```

Comment:
Minor things, but I would rewrite the code like so:
```c#
if (query is CollectionNames)
{
    List<string> collectionNames = new List<string>();
    if (m_Database != null)
    {
        foreach (BsonDocument doc in m_Database.ListCollectionsAsync().Result.ToListAsync<BsonDocument>().Result)
        {
            if (doc.Contains("name"))
                collectionNames.Add(doc["name"].AsString);
        }
    }
    return collectionNames;
}
```

3 main points in there:
- return an empty list instead of null
- check that the bson document contains a name otherwise, it will throw an exception
- better to put curly brackets for the `if` when there is more than one line to make it easier to read